test/saithrift/tests/copp_tests.py

```diff
@@ -0,0 +1,549 @@
+        sai_thrift_create_hostif_trap(
+            client=self.client, 
+            trap_id=SAI_HOSTIF_TRAP_ID_ARP_RESPONSE, 
+            action=SAI_PACKET_ACTION_TRAP, 
+            channel=SAI_HOSTIF_TRAP_CHANNEL_NETDEV, 
+            trap_group_id=trap_group)
+
+        #'dhcp,lldp q4 no policer'
+        trap_group = sai_thrift_create_hostif_trap_group(self.client, queue_id=4)
+        self.trap_groups.append(trap_group)
+        sai_thrift_create_hostif_trap(
+            client=self.client, 
+            trap_id=SAI_HOSTIF_TRAP_ID_LLDP, 
+            action=SAI_PACKET_ACTION_TRAP, 
+            channel=SAI_HOSTIF_TRAP_CHANNEL_NETDEV, 
+            trap_group_id=trap_group)
+        sai_thrift_create_hostif_trap(
+            client=self.client, 
+            trap_id=SAI_HOSTIF_TRAP_ID_DHCP, 
+            action=SAI_PACKET_ACTION_TRAP, 
+            channel=SAI_HOSTIF_TRAP_CHANNEL_NETDEV, 
+            trap_group_id=trap_group)
+
+        #'ip2me q1 policer '
+        sai_policer_id = sai_thrift_create_policer(self.client, 
+                            meter_type=SAI_POLICER_MODE_Sr_TCM, 
+                            mode=SAI_METER_TYPE_PACKETS,
+                            cir=self.POLICER_CIR,
+                            red_action=SAI_PACKET_ACTION_DROP)
+        self.policers.append(sai_policer_id)
+
+        trap_group = sai_thrift_create_hostif_trap_group(self.client, queue_id=4, policer_id=sai_policer_id)
+        self.trap_groups.append(trap_group)
+        sai_thrift_create_hostif_trap(
+            client=self.client, 
+            trap_id=SAI_HOSTIF_TRAP_ID_IP2ME,
+            action=SAI_PACKET_ACTION_TRAP, 
+            channel=SAI_HOSTIF_TRAP_CHANNEL_NETDEV, 
+            trap_group_id=trap_group)
+        return
+
+    def create_routes(self):
+        for i in xrange(self.MAX_PORTS):
+            self.client.sai_thrift_set_if_state('Ethernet%d' % (i+1), True)
+            self.setup_route(self.peerip[i])
+            sai_port_id = self.client.sai_thrift_get_port_id_by_front_port('Ethernet%d' % (i+1));
+            self.setup_direct_route(sai_port_id, self.peerip[i])
+        return
+
+    def init_interfaces(self):
+        self.loadPortMap()
+        self.createRpcClient()            
+        switch_init(self.client)
+        self.create_host_interfaces()
+        self.create_routes()
+        return
+
+    def __init__(self):
+
+        BaseTest.__init__(self)
+        
+        self.test_params = testutils.test_params_get()
+        
+        self.mac_map = []
+        for i in xrange(self.MAX_PORTS):
+            output = ControlPlaneBaseTest.cmd_run('ip link show dev eth%d' % (i))
+            second = output.split('\n')[1]
+            mac = second.split()[1]
+            self.mac_map.append(mac)
+            
+        self.myip = {}
+        self.peerip = {}
+        for i in xrange(self.MAX_PORTS):
+            self.myip[i] = "10.0.0.%d" % (i*2+1)
+            self.peerip[i] = "10.0.0.%d" % (i*2)
```

Comment:
Hi Guohan, based on conversation we had yesterday evening, it does what is expected. Please let's have another look, and if there are question, let's meet in person and close the issue.